cloudtracker/__init__.py

```diff
@@ -526,8 +526,19 @@ def run(args, config, boto3_session, start, end, account_iam, datasource):
 
     if not account_iam:
         account_iam = get_account_iam(account, boto3_session)
-    users_performed_actions = {}
-    roles_performed_actions = {}
+
+    search_query = datasource.get_search_query()
+
+    if args[0].user:
+        if args[0].destpolicy:
+            users_performed_actions = datasource.get_performed_event_names_by_users(search_query, principals_arn)
+        elif args[0].permissionsetid:
+            users_performed_actions = datasource.get_performed_event_names_by_sso_users(search_query, principals_arn)
+    elif args[0].role:
+        roles_performed_actions = datasource.get_performed_event_names_by_roles(search_query, principals_arn)
+    else:
```

Comment:
instead of assigning variables inside `else`, let's assign these before the `if` condition